src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/AccountSupervisor.kt

```diff
@@ -767,20 +769,29 @@ internal open class AccountSupervisor(
             val result = helper.parser.decodeJsonObject(additionalPayload)
 
             if (error == null && result != null) {
-                val url = complianceGeoblockUrl()
                 val signedMessage = helper.parser.asString(result["signedMessage"])
                 val publicKey = helper.parser.asString(result["publicKey"])
                 val timestamp = helper.parser.asString(result["timestamp"])
+                val isKeplr = helper.parser.asBool(result["isKeplr"])
+                val url = if (isKeplr == true) complianceGeoblockKeplrUrl() else complianceGeoblockUrl()
```

Comment:
[discussion] we're starting to get a lot of branching in this function. you think it's worth splitting this into 2 separate functions? or creating 2 separate helper functions?

if this is urgent i'm down to stamp right away though since the logic makes sense

Reply:
Definitely thought about it, and tried initially. Ended with this approach because the only code path change is just the body and url, but if you feel strongly I can take another stab!

Reply:
nah no worries! i think we can do this as part of a larger cleanup task in the future to reduce code duplication. the `OnboardingSupervisor` is a really bad offender of this... so easy to make mistakes when updating things since there's so much duplicate code

Reply:
yea, feel that